server/src/HatoholArmPluginGateHAPI2.cc

```diff
@@ -157,6 +177,69 @@ struct HatoholArmPluginGateHAPI2::Impl
 		m_armStatus.setRunningStatus(false);
 	}
 
+	void setupSelfMonitors(void)
+	{
+		auto ignore = [](SelfMonitor &m, ...) {};
+		auto log = [](SelfMonitor &m, ...) {
+			 m.saveEvent();
+		};
+		auto check = [&](SelfMonitor &monitor,
+		                 const TriggerStatusType &prevStatus,
+		                 const TriggerStatusType &currStatus) {
+			if (monitor.getLastKnownStatus() != currStatus)
+				log(monitor, prevStatus, currStatus);
+		};
+		const SelfMonitor::EventGenerator
+		  mtx[NUM_TRIGGER_STATUS][NUM_TRIGGER_STATUS] = {
+			{
+				ignore, /* OK -> OK */
+				log,    /* OK -> PROBLEM */
+				ignore, /* OK -> UNKNOWN */
+			}, {
+				log,    /* PROBLEM -> OK */
+				log,    /* PROBLEM -> PROBLEM */
+				ignore, /* PROBLEM -> UNKNOWN */
+			}, {
+				check,  /* UNKNOWN -> OK */
+				check,  /* UNKNOWN -> PROBLEM */
+				ignore, /* UNKNOWN -> UNKNOWN */
+			}
+		};
+
+		auto setAllStatusComb = [&](SelfMonitorPtr m) {
+			for (int p = 0; p < NUM_TRIGGER_STATUS; p++) {
+				for (int c = 0; c < NUM_TRIGGER_STATUS; c++) {
+					m->setEventGenerator(
+					  static_cast<TriggerStatusType>(p),
+					  static_cast<TriggerStatusType>(c),
+					  mtx[p][c]);
+				}
+			}
+		};
+
```

Comment:
I didn't want to break lines to keep good visibility. So I used those short names.
However, some people would think like you.
I'll use proper variable names with readable code structure.